### ISSUE TITLE
How to disable Compiler diagnostics in oclint report?

### ISSUE THREAD

**Issue by cuixuejiao** [oclint/oclint#481]
### Issue Summary
Basically, Summary, A list of violations, A list of violations, and OCLint information would be found in the oclint report. Bug I don't want compiler errors and compiler warnings in my report.

### Environment
- Operation system name and version: 10.12.6
- OCLint version: 0.13
- How OCLint is installed: prebuilt binary downloaded from github

### Reproduction Steps
oclint-json-compilation-database \
                        -i ${LINT_INCLUDES} \
                        -e ${LINT_EXCLUDES} \
                        -- -report-type ${LINT_REPORT_TYPE} \
                           ${LINT_RULES} \
                           ${LINT_THRESHOLD} \
                           -o ${LINT_REPORT_FILE} \
                           -stats \
                           -verbose \
                           -list-enabled-rules \
                           -no-analytics

#### Sample code

```
Compiler Warnings:

xxx/Manager+AppServer.h:52:12: parameter 'deviceLevel' not found in the function declaration

```

Command to run OCLint with the code above.

### Expected Behavior
Skip Compiler diagnostics in oclint report, or don't show Compiler diagnostics in oclint report.

### Actual Behavior
Compiler diagnostics shows in oclint report.

Please help me out, thank you~

**Comment by UnrealQuester** [comment]
oclint takes the warnings directly from the compiler. If you do not want these warnings in your report, you cannot have warnings enabled in your compile flags.

You can either make sure that your compile commands do not contain any warning flags or disable them by adding `-extra-arg=-Wno-everything` to the oclint flags. You can append the option like this:
`oclint-json-compilation-database <YOUR OPTIONS HERE> -- -extra-arg=-Wno-everything`

**Comment by cuixuejiao** [comment]
-extra-arg=-Wno-everything

This parameter works! Thank you so much~@UnrealQuester

**Comment by greenzeal** [comment]
still compiler errors are in the report

**Comment by fndppx** [comment]
> oclint takes the warnings directly from the compiler. If you do not want these warnings in your report, you cannot have warnings enabled in your compile flags.
> 
> You can either make sure that your compile commands do not contain any warning flags or disable them by adding `-extra-arg=-Wno-everything` to the oclint flags. You can append the option like this: `oclint-json-compilation-database <YOUR OPTIONS HERE> -- -extra-arg=-Wno-everything`

Why didn't it work in my project

<img width="1070" alt="image" src="https://user-images.githubusercontent.com/7971897/160308125-d9404d16-1dbd-4fba-8db8-30a044eccbab.png">

OCLint got compile errors in html report file, but my project build success. 

I found that Oclint did not recognize the PCH import header file

**Comment by fndppx** [comment]
> -extra-arg=-Wno-everything
> 
> This parameter works! Thank you so much~@UnrealQuester

Why didn't it work in my project
